addons/mail/static/tests/core/record.test.js

```diff
@@ -961,6 +961,125 @@ test("record.toData() is JSON stringified and can be reinserted as record", asyn
     expect(serializeDateTime(p2.due_datetime)).toBe("2024-08-28 10:19:44");
 });
 
+test("record.toData() returns flat data", async () => {
+    (class Person extends Record {
+        static id = "id";
+        id;
+        names = Record.attr([]);
+        due_datetime = Record.attr(undefined, { type: "datetime" });
+        messages = Record.many("Message");
+        team = Record.one("Team");
+    }).register(localRegistry);
+    (class Message extends Record {
+        static id = "id";
+        id;
+        body = Record.attr("");
+    }).register(localRegistry);
+    (class Team extends Record {
+        static id = "id";
+        id;
+        name;
+        leader = Record.one("Person");
+    }).register(localRegistry);
+    const store = await start();
+    store.Person.insert([
+        {
+            id: 1,
+            due_datetime: "2024-08-28 10:19:44",
+            names: ["John", "Marc"],
+            messages: [
+                { id: 1, body: "1" },
+                { id: 2, body: "2" },
+            ],
+            team: { id: 1, name: "Discuss", leader: { id: 2 } },
+        },
+        {
+            id: 2,
+            due_datetime: "2025-01-23 12:12:12",
+            names: ["Alex"],
+            messages: [
+                { id: 1, body: "1" },
+                { id: 3, body: "3" },
+            ],
+            team: { id: 2, name: "VoIP", leader: { id: 1 } },
+        },
+    ]);
```

Comment:
In practice Louis is the VOIP leader/guru and Seb the Discuss leader/guru. Don't know if they have more than 1 name, my 2nd name is "Erick" by the way 😆

Reply:
ok, Erick 😆